src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions_Tracking.cs

```diff
@@ -155,7 +158,8 @@ private static SyntaxNode GetRoot(SyntaxNode node)
                 }
                 else
                 {
-                    node = ((IStructuredTriviaSyntax)node).ParentTrivia.Token.Parent;
+                    node = ((IStructuredTriviaSyntax)node).ParentTrivia.Token.Parent!;
```

Comment:
Consider leaving comment to use MemberNotNull on IsStructuredTrivia

Reply:
Don't think `MemberNotNull` will help here. The member which is `null` here is the final `.Parent` call. That isn't a direct member of `node` hence it can't be updated via a `MemberNotNull` on `IsStructuredTrivia`